src/ssh/azext_ssh/target_properties_utils.py

```diff
@@ -0,0 +1,203 @@
+        if properties.get('properties', None):
+            return properties.get('properties').get('osType', None)
+
+    elif resource_type == "microsoft.connectedvmwarevsphere/virtualmachines":
+        if properties.get("osProfile") and properties.get("osProfile").get("osType"):
+            return properties.get("osProfile").get("osType")
+
+
+def parse_agent_version(properties, resource_type):
+    if resource_type == "microsoft.compute/virtualmachines":
+        return None
+
+    if resource_type == "microsoft.hybridcompute/machines":
+        if properties.get('properties'):
+            return properties.get('properties').get('agentVersion')
+
+    if resource_type == "microsoft.connectedvmwarevsphere/virtualmachines":
+        if properties.get("properties") and properties.get("properties").get('guestAgentProfile') and\
+                properties.get("properties").get('guestAgentProfile').get('agentVersion'):
+
+            return properties.get("properties").get('guestAgentProfile').get('agentVersion')
+
+
+def parse_tags(properties, resource_type):
+    if resource_type == "microsoft.compute/virtualmachines":
+        if properties.tags:
+            return properties.tags
+
+    else:
+        if properties.get('tags'):
+            return properties.get('tags')
+
+
+# This function is used to check if the OS type is valid and if the authentication options are valid for that OS
+def check_valid_os_type(os_type, op_info):
+    if os_type:
+        logger.debug("Target OS Type: %s", os_type)
+        telemetry.add_extension_event('ssh', {'Context.Default.AzureCLI.TargetOSType': os_type})
+
+    if os_type and os_type.lower() == 'windows' and not op_info.local_user:
+        colorama.init()
+        error_message = "SSH Login using AAD credentials is not currently supported for Windows."
+        recommendation = colorama.Fore.YELLOW + "Please provide --local-user." + colorama.Style.RESET_ALL
+        raise azclierror.RequiredArgumentMissingError(error_message, recommendation)
+
+
+def check_valid_agent_version(agent_version, op_info):
+    # pylint: disable=broad-except
+    if op_info.is_arc() and agent_version:
+        try:
+            major, minor, _, _ = agent_version.split('.', 4)
+            if int(major) < const.AGENT_MINIMUM_VERSION_MAJOR or int(minor) < const.AGENT_MINIMUM_VERSION_MINOR:
+                logger.warning("The version of the Arc Agent, %s running on the target machine "
+                               "is not compatible with this version of the ssh extension. "
+                               "Please update to the latest version.", agent_version)
+        except Exception:
+            return
+
+
+# This function is used to set the resource tags and configure the port number from the resource tag
+def handle_resource_tags_utils(tags, op_info):
+    set_azure_resource_tags(tags, op_info)
+    configure_port_from_resource_tag(op_info)
+
+
+def set_azure_resource_tags(tags, op_info):
```

Comment:
This function is so tiny, I think you can just write that logic under handle_resource_tags_utils